### PR TITLE
Add model.eval() in train.py

### PR DESCRIPTION
**Pull request by karandwivedi42**

### CODE REVIEW

**Comment by adampolyak** [comment]
Thanks for the PR. 
Eval mode is used only during inference, by disabling teacher forcing. During test time, teacher forcing is **enabled** in order to have test and train scores comparable.
